staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/marshal_test.go

```diff
@@ -18,133 +18,522 @@ package v1beta1
 
 import (
 	"encoding/json"
-	"reflect"
+	"math"
 	"testing"
+
+	"github.com/fxamacker/cbor/v2"
+	"github.com/google/go-cmp/cmp"
 )
 
-type JSONSchemaPropsOrBoolHolder struct {
-	JSPoB          JSONSchemaPropsOrBool  `json:"val1"`
-	JSPoBOmitEmpty *JSONSchemaPropsOrBool `json:"val2,omitempty"`
+type marshalTestable interface {
+	json.Marshaler
+	cbor.Marshaler
 }
 
-func TestJSONSchemaPropsOrBoolUnmarshalJSON(t *testing.T) {
-	cases := []struct {
-		input  string
-		result JSONSchemaPropsOrBoolHolder
-	}{
-		{`{}`, JSONSchemaPropsOrBoolHolder{}},
+type marshalTestCase struct {
+	name          string
+	input         marshalTestable
+	wantJSONError bool
+	wantCBORError bool
+	wantJSON      []byte
+	wantCBOR      []byte
+}
 
-		{`{"val1": {}}`, JSONSchemaPropsOrBoolHolder{JSPoB: JSONSchemaPropsOrBool{Allows: true, Schema: &JSONSchemaProps{}}}},
```

Comment:
it's a little hard to tell with the rewrite... did the existing test cases get preserved? is there a way to make sure the pre-existing unit tests pass as-is against the non-test changes in this PR?

Reply:
Kicked off https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/127482/pull-kubernetes-unit/1836853101936513024 with the original tests intact (job just started, but "works on my machine").

Reply:
Here's a coverage comparison between master and this PR. This metric is at least as good or better than master, plus, the methods on master with no existing test coverage are now covered:

```
Receiver				Method		Old	New

v1.JSON					MarshalJSON	100.0%	100.0%
v1.JSON					UnmarshalJSON	100.0%	100.0%
v1.JSONSchemaPropsOrArray		MarshalJSON	100.0%	100.0%
v1.JSONSchemaPropsOrArray		UnmarshalJSON	85.7%	85.7%
v1.JSONSchemaPropsOrBool		MarshalJSON	100.0%	100.0%
v1.JSONSchemaPropsOrBool		UnmarshalJSON	83.3%	91.7%
v1.JSONSchemaPropsOrStringArray		MarshalJSON	0.0%	100.0%
v1.JSONSchemaPropsOrStringArray		UnmarshalJSON	0.0%	85.7%
v1beta1.JSON				MarshalJSON	100.0%	100.0%
v1beta1.JSON				UnmarshalJSON	100.0%	100.0%
v1beta1.JSONSchemaPropsOrArray		MarshalJSON	100.0%	100.0%
v1beta1.JSONSchemaPropsOrArray		UnmarshalJSON	85.7%	85.7%
v1beta1.JSONSchemaPropsOrBool		MarshalJSON	100.0%	100.0%
v1beta1.JSONSchemaPropsOrBool		UnmarshalJSON	83.3%	91.7%
v1beta1.JSONSchemaPropsOrStringArray	MarshalJSON	0.0%	100.0%
v1beta1.JSONSchemaPropsOrStringArray	UnmarshalJSON	0.0%	85.7%
```

Reply:
thanks, the combination of those two bits sgtm